ml-conversational-analytic-tool/tests/test_commentAnalysis.py

```diff
@@ -0,0 +1,64 @@
+import unittest
+
+from ..commentAnalysis import CommentAnalyzer
+
+class TestCommentAnalysis(unittest.TestCase):
+
+    def test_CommentAnalyzer_init(self):
+        analyzer = CommentAnalyzer(['test'])
+        self.assertEqual({'test': 0}, analyzer.word_count,)
+
+    def test_analyzeComment(self):
+        analyzer = CommentAnalyzer(['test'])
+        result = analyzer.analyzeComment("This is a test PR")
+        self.assertEqual({'test': 1, 'Sentiment': 0.0, 'Code Blocks': 0}, result)
+
+    def test_analyzeComment_with_empty_comment(self):
+        analyzer = CommentAnalyzer(['empty comment'])
+        result = analyzer.analyzeComment("")
+        self.assertEqual({'empty comment': 0, 'Sentiment': 0.0, 'Code Blocks': 0}, result)
+
+    def test_analyzeComment_positive_comment(self):
+        analyzer = CommentAnalyzer(['positive comment'])
+        result = analyzer.analyzeComment("This patch is really good")
+        self.assertEqual({'positive comment': 0, 'Sentiment': 0.4927, 'Code Blocks': 0}, result)
+
+    def test_analyzeComment_negative_comment(self):
+        analyzer = CommentAnalyzer(['negative comment'])
+        result = analyzer.analyzeComment("This patch is bad")
+        self.assertEqual({'negative comment': 0, 'Sentiment': -0.5423, 'Code Blocks': 0}, result)
+
+    def test_analyzeComment_with_codeBlocks(self):
+        analyzer = CommentAnalyzer(['code block test'])
+        result = analyzer.analyzeComment("```This patch has blocks```, ```This is second block```")
+        self.assertEqual({'code block test': 0, 'Sentiment': 0.0, 'Code Blocks': 2}, result)
+
+    def test_preProcess(self):
+        analyzer = CommentAnalyzer(['preProcess test'])
+        result = analyzer.preProcess("THIS is a TEST COMMENT")
+        self.assertEqual("this is a test comment", result)
+
+    def test_countWords(self):
+        analyzer = CommentAnalyzer(['count words test'])
+        result = analyzer.countWords("This is a test comment")
+        self.assertEqual({'count words test': 0}, result)
+
+    def test_getSentiment_positive(self):
+        analyzer = CommentAnalyzer(['positive sentiment test'])
+        result = analyzer.getSentiment("This is a very good PR")
+        self.assertEqual(0.4927, result)
+
+    def test_getSentiment_neutral(self):
+        analyzer = CommentAnalyzer(['neutral sentiment test'])
+        result = analyzer.getSentiment("This is a decent PR")
+        self.assertEqual(0.0, result)
+
+    def test_getSentiment_negative(self):
+        analyzer = CommentAnalyzer(['negative sentiment test'])
+        result = analyzer.getSentiment("This is a very bad PR")
+        self.assertEqual(-0.5849, result)
```

Comment:
I don't think these tests are needed as sentiment depends solely on what Vader returns.
Rather, we should focus test on our functionality which is that it returns a float.

Reply:
Just to clarify, my [initial comment](https://github.com/vmware-labs/ml-conversational-analytic-tool/pull/17#discussion_r778430314) was related to how tests should not always have sentiment and code blocks as 0.

I think using mock for sentiment value as specified in https://github.com/vmware-labs/ml-conversational-analytic-tool/pull/17#discussion_r782478864 and having test related to https://github.com/vmware-labs/ml-conversational-analytic-tool/pull/17#discussion_r778425113 would meet that.

If anything is confusing, let me know! thanks!